go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -692,3 +695,161 @@ func main() {
 
 	extract(depMode, modMode)
 }
+
+const minGoVersion = "1.11"
+const maxGoVersion = "1.20"
+
+func outsideSupportedRange(version string) bool {
+	short := semver.MajorMinor("v" + version)
+	return semver.Compare(short, "v"+minGoVersion) < 0 || semver.Compare(short, "v"+maxGoVersion) > 0
+}
+
+func checkForUnsupportedVersions(v versionInfo) (msg, version string) {
+	if v.goDirectiveFound && outsideSupportedRange(v.goModVersion) {
+		msg = "The version of Go found in the `go.mod` file (" + v.goModVersion +
+			") is outside of the supported range (" + minGoVersion + "-" + maxGoVersion + ")."
+		version = ""
+		diagnostics.EmitUnsupportedVersionGoMod(msg)
+	}
+
+	if v.goInstallationFound && outsideSupportedRange(v.goEnvVersion) {
+		msg = "The version of Go installed in the environment (" + v.goEnvVersion +
+			") is outside of the supported range (" + minGoVersion + "-" + maxGoVersion + ")."
+		version = ""
+		diagnostics.EmitUnsupportedVersionEnvironment(msg)
+	}
+
+	return msg, version
+}
+
+func checkForVersionsNotFound(v versionInfo) (msg, version string) {
+	if !v.goInstallationFound && !v.goDirectiveFound {
+		msg = "No version of Go installed and no `go.mod` file found. Writing an " +
+			"`environment.json` file specifying the maximum supported version of Go (" +
```

Comment:
For all the occurrences of `environment.json` here, you can probably use a generic name here.